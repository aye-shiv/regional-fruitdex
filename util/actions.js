export const GET_FRUITS = 'GET_FRUITS';

const API_URL = 'https://fruitdexapinodejs.herokuapp.com/fruits';

export const getFruit = (path, options = {}) => {
    try {
        return async dispatch => {
            const res = await(fetch(API_URL, {
                method: "GET",
                headers: {
                    'Content-Type': 'application/json',
                }
            }));
            const json = await(res.json());
            if(json){
                dispatch({
                    type: GET_FRUITS,
                    payload: json.fruits
                });
            } else {
                console.log('Failed fetch');
            }
        }
    } catch (err) {
        console.log(err);
    }
}