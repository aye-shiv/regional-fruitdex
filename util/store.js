import { createStore, applyMiddleware, combineReducers } from 'redux';
import ReduxThunk from 'redux-thunk';
import {persistReducer, persistStore} from 'redux-persist'
import AsyncStorage from '@react-native-async-storage/async-storage';
import dexReducer from './reducers'

const persistConfig = {
    key: 'persist-key',
    storage: AsyncStorage
}

const rootReducer = combineReducers({
    dexReducer: persistReducer(persistConfig,dexReducer)
});

export const store = createStore(rootReducer, applyMiddleware(ReduxThunk));
export const persistor = persistStore(store);