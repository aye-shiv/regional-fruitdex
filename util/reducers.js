import { GET_FRUITS } from "./actions";
import { combineReducers } from 'redux';

const initialState = {
    fruits: []
}

function dexReducer(state = initialState, action){
    switch (action.type){
        case GET_FRUITS:
            return {...state, fruits: action.payload};
        default:
            return state;
    }
}
export default dexReducer;