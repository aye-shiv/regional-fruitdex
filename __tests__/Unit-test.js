import dexReducer from '../util/reducers';
import { getFruit } from '../util/actions';
import GET_FRUITS from '../util/actions';


describe('dexReducer', () => {
  it('returns the initial state', () => {
    expect(dexReducer(undefined, {})).toMatchSnapshot();
  });
  it('handles the getFruit action', () => {
    expect(dexReducer({}, getFruit())).toMatchSnapshot();
  });
});



