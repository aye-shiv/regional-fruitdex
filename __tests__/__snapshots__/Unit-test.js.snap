// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dexReducer handles the getFruit action 1`] = `Object {}`;

exports[`dexReducer returns the initial state 1`] = `
Object {
  "fruits": Array [],
}
`;
